perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -518,6 +592,178 @@ def SingleStreamThroughputBenchmark(storage_schema, host_to_connect=None):
                   host_to_connect=host_to_connect)
 
 
+def RunMultiThreadedWorkersWriteResults(worker, worker_args,
+                                        storage_schema, host_to_connect=None):
+  """Run a worker function in many threads, then gather and print the results.
+
+  Args:
+    worker: either WriteWorker or ReadWorker. The worker function to call.
+    worker_args: a tuple. The arguments to pass to the worker function. The
+      result queue and stream number will be appended as the last two arguments.
+    storage_schema: The schema of the storage provider to use, e.g., "gs"
+    host_to_connect: An optional host endpoint to connect to.
+
+  The object size is given by FLAGS.object_size_kb. The number of
+  objects to transfer (per stream) is given by
+  FLAGS.objects_per_stream. The benchmark starts at FLAGS.start_time,
+  if given, or else immediately.
+
+  Raises:
+    LowAvailabilityError: when the storage provider has failed a high number of
+        our RW requests that exceeds a threshold (>5%), we raise this error
+        instead of collecting performance numbers from this run.
+  """
+
+  result_queue = Queue.Queue()
+
+  logging.info('Creating threads')
+  threads = [threading.Thread(target=worker,
+                              args=worker_args + (result_queue, i))
+             for i in xrange(FLAGS.num_streams)]
+  logging.info('Threads created. Starting threads.')
+  for thread in threads:
+    thread.start()
+  logging.info('Threads started. Joining threads.')
+  for thread in threads:
+    thread.join()
+  logging.info('All threads complete.')
+
+  try:
+    num_results = 0
+    while True:
+      res = result_queue.get_nowait()
+      num_results += 1
+      stream_num = res['stream_num']
+      for start_time, latency in zip(res['start_times'], res['latencies']):
+        json.dump({'kind': res['kind'],
+                   'start_time': start_time,
+                   'latency': latency,
+                   'size': FLAGS.object_size_kb,
+                   'stream_num': stream_num},
+                  sys.stdout)
+        sys.stdout.write('\n')
+  except Queue.Empty:
+    pass
+
+  if num_results != FLAGS.num_streams:
+    raise Exception('Couldn\'t get results from all streams!')
```

Comment:
Yes, I'll just return the results. I don't think we have to pass back num_results because the results already include stream numbers, so the caller can figure out how many streams are there.